app/src/main/java/com/example/easycloset/HomeFragment.java

```diff
@@ -55,5 +78,58 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         tvMaxTemp = view.findViewById(R.id.tvMaxTemp);
         tvSunrise = view.findViewById(R.id.tvSunrises);
         tvSunset = view.findViewById(R.id.tvSunsets);
+        nestedScrollView = view.findViewById(R.id.homeScreen);
+    }
+
+    public void setWeather(Weather weather) {
+        this.weather = weather;
+    }
+
+    public void setLatitude(double latitude) {
+        this.latitude = latitude;
+    }
+
+    public void setLongitude(double longitude) {
+        this.longitude = longitude;
+    }
+
+    public double getLatitude() {
+        return latitude;
+    }
+
+    public double getLongitude() {
+        return longitude;
+    }
+
+    public void populateHomeFragment() {
```

Comment:
the name is a bit confusing, should be something like "fetchWeatherData"

Reply:
fixed